overlord/snapstate/snapstate.go

```diff
@@ -1968,6 +1971,20 @@ func autoRefreshPhase1(ctx context.Context, st *state.State) ([]string, []*state
 		}
 	}
 
+	if forGatingSnap != "" {
+		var gatingSnapHasUpdate bool
+		for _, up := range updates {
+			if up.InstanceName() == forGatingSnap {
+				gatingSnapHasUpdate = true
+				break
+			}
+		}
+		if !gatingSnapHasUpdate {
+			// XXX: should this error out instead?
```

Comment:
this seems fine, as we anyway error here https://github.com/snapcore/snapd/pull/10515/files#diff-61e060364d4189cc8f03d8118c72257b2ce915a1759a83a6b5961340b758507bR660 which is checking the more concrete condition